lib/gather-fragment-imports.js

```diff
@@ -0,0 +1,58 @@
+const path = require('path');
+
+const inlineImports = require('@graphql-fragment-import/lib/inline-imports');
+
+/**
+ * Helper function to fetch and parse all the direct and transitive imports starting
+ * from the source. Return object is:
+ *
+ * {
+ *     1: {
+ *         FooBar: fragment,
+ *         Bar: fragment
+ *     }
+ * }
+ *
+ * @param {string} source - .graphql source code
+ * @param {string} sourceLocation - File path of the source code
+ * @param {function} resolveImport - This function is used to resolve the path to file for import
+ * @param {function} fragmentParserGenerator - Generator function that is passed source code. Expected to yield objects for each fragment definition
+ * @param {boolean} throwIfImportNotFound - If set to true, throws error if import not found
+ */
+function gatherFragmentImports({
+  source,
+  sourceLocation,
+  resolveImport,
+  fragmentParserGenerator,
+  throwIfImportNotFound,
+}) {
+  /**
+   * {
+   *     1: {
+   *         FooBar: fragment,
+   *         Bar: fragment
+   *     }
+   * }
+   * @type {{number: { string: array<object>}}}
+   */
+  const lineToFragmentDefinitions = {};
```

Comment:
Why are we returning an object instead of a `Map`? Shouldn't we be consistent with `lineToImports`?

Reply:
used map